packages/core/src/server/index.ts

```diff
@@ -189,12 +119,12 @@ export async function createServer({
     const httpServer = serve(
       {
         fetch: hono.fetch,
-        createServer: createServerWithNextAvailablePort,
-        port,
+        createServer: http.createServer,
+        port: apiBuild.port,
```

Comment:
Fine to ship, but flagging again that this can fail here if the port has been taken between build and this line.